docs/index.rst

```diff
@@ -155,14 +171,115 @@ Examining the content of the Python, a few things to note are;
+
+If you no wish one filter to pass data onto a later filter in the pipeline this is best done by adding data to the reading, as an extra data point. This data point can then be removed by the later filter. An example of this is the way Fledge adds OMF hints to readings that are processed and removed by the OMF north plugin.
+
+For example let us assume we have calculated some value *delta* that we wish to pass to a later filter, we can add this as a data point to our reading which we will call *_hintDelata*.
+
+.. code-block:: python
+
+    
+   def myPython(readings):
+       for elem in list(readings):
+           reading = elem['readings']
+           ...
+           reading['_hintDelta'] = delta
+           ...
+       return readings
+
+This is far better than using a global as it is attached to the reading to which it refers and will remain attached to that reading until it is removed. It also means that it is independent of the number of readings that are processed per call, and resilient to readings being added or removed from the stream.
+
+The name chosen for this data point in the example above has no significance, however it is good practice to choose a name that is unlikely to occur in the data normally and portrays the usage or meaning of the data.
+
+File IO Operations
+~~~~~~~~~~~~~~~~~~
+
+It is possible to make use of file operations within a Python35 filter function, however it is not recommended for production use for the following reasons; 
+
+  - Pipelines may be moved to other hosts where files may not be accessible.
+    
+  - Permissions may change dependent upon how Fledge systems are deployed in the various different scenarios.
+   
+  -  Edge devices may also not have large, high performance storage available, resulting in performance issues for Fledge or failure due to lack of space. 
+
+  - Fledge is designed to be managed solely via the Fledge API and applications that use that API. There is no facility within that API to manage arbitrary files within the filesystem.
+
+It is common to make use of files during development of a script to write information to in order to aid development and debugging, however this should be removed, along with associated imports of packages required to perform the file IO, when a filter is put into production.
+
+Threads within Python
+~~~~~~~~~~~~~~~~~~~~~
+
+It is tempting to use threads within Python to perform background activity or to allow processing of data sets in parallel, however there is an issue with threading in Python, the Python Global Interpreter Lock or GIL. The GIL prevents two Python statements from being executed within the same interpreter by two threads simultaneously. Because we use a single interpreter for all Python code running in each service within Fledge, if a Python thread is created that performs CPU intensive work within it, we block all other Python code from running within that Fledge service.
```

Comment:
It seems something is missing between "however there is an issue with threading in Python, " and "the Python Global Interpreter Lock or GIL"

Reply:
No, nothing is missing.